### PR TITLE
BestSalesProductSearchProvider ignores sorting argument

### LINKED ISSUE #21600: BestSalesProductSearchProvider ignores sorting argument when 'order' query param is falsy
#### Describe the bug

I am trying to display best selling products from a module. For that I am leveraging `BestSalesProductSearchProvider` with random sort order. Even if I provide sorting order, it still defaults to number of sales. This is because of the implementation, which ignores provided sorting order and sets the sorting order to number of sales if `order` query parameter is falsy:

https://github.com/PrestaShop/PrestaShop/blob/1.7.6.8/src/Adapter/BestSales/BestSalesProductSearchProvider.php#L72-L74

Why is there such restriction? If I want to display best selling products block on homepage, then I cannot image having an `order` query parameter attached.

#### Expected behavior

`BestSalesProductSearchProvider` returns products in order defined by sorting parameter.

#### Steps to Reproduce

Use the following code to get raw best selling products:
```
$searchProvider = new BestSalesProductSearchProvider(
   $this->context->getTranslator()
);

$context = new ProductSearchContext($this->context);
$query = new ProductSearchQuery();
$nProducts = 20;
$query
   ->setResultsPerPage($nProducts)
   ->setPage(1);
$query->setSortOrder(SortOrder::random()); //this gets ignored

$result = $searchProvider->runQuery(
   $context,
   $query
);
```
#### Additional information

* PrestaShop version: 1.7.6.3 & 1.7.7.0
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by ghost**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | I am trying to display best selling products from a module. For that I am leveraging BestSalesProductSearchProvider with random sort order. Even if I provide sorting order, it still defaults to number of sales.
| Type?             | improvement
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/21600.
| How to test?      | Install this module and check if random work
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24349)

### CODE REVIEW

File: classes/ProductSale.php [hunk 1]
```diff
@@ -96,6 +96,12 @@ public static function getBestSales($idLang, $pageNumber = 0, $nbProducts = 10,
         if ($invalidOrderWay || null === $orderWay || $orderBy == 'sales') {
             $orderWay = 'DESC';
         }
+        
+        if ($orderWay == 'random') {
```
Comment (Progi1984) [inline classes/ProductSale.php]:
```suggestion
        if ($orderWay === 'random') {
```

File: classes/Validate.php [hunk 2]
```diff
@@ -761,7 +761,7 @@ public static function isZipCodeFormat($zip_code)
      */
     public static function isOrderWay($way)
     {
-        return $way === 'ASC' | $way === 'DESC' | $way === 'asc' | $way === 'desc';
+        return $way === 'ASC' | $way === 'DESC' | $way === 'asc' | $way === 'desc' | $way === 'random';
```
Comment (Progi1984) [inline classes/Validate.php:764]: Could you add unit tests ;) ?

File: src/Adapter/BestSales/BestSalesProductSearchProvider.php [hunk 3]
```diff
@@ -69,7 +69,7 @@ public function runQuery(
             $this->translator->trans('Sales, highest to lowest', [], 'Shop.Theme.Catalog')
         );
 
-        if (!Tools::getValue('order', 0)) {
+        if (!Tools::getValue('order', 0) && $query->getSortOrder()->toLegacyOrderWay() != 'random') {
```
Comment (Progi1984) [inline src/Adapter/BestSales/BestSalesProductSearchProvider.php]:
```suggestion
        if (!Tools::getValue('order', 0) && $query->getSortOrder()->toLegacyOrderWay() !== 'random') {
```

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

File: classes/Validate.php (hunk 2 above)
Comment (ghost) [reply classes/Validate.php:764]: It's a good oportunity :)

**Comment by ghost** [comment]
Hey @Progi1984 i try to update `classes/ProductSale.php` but my cs-fixer don't change anything and when i try ti use .php_cs.dist from prestashop i have an error ( my envorinement is not very good ) it's possible for you to run cs-fixer on it ? Sorry !

**Pull request closed without merging**